replayserver/bookkeeping/bookkeeper.py

```diff
@@ -34,13 +34,15 @@ async def save_replay(self, game_id, stream):
             await self._saver.save_replay(game_id, stream)
             logger.debug(f"Saved replay {game_id}")
             metrics.saved_replays.inc()
+            replay_available = True
         except BookkeepingError as e:
             logger.warning(f"Failed to save replay for game {game_id}: {e}")
+            replay_available = False
 
         try:
             logger.debug(f"Analyzing replay {game_id}")
             ticks = self._analyzer.get_replay_ticks(stream.data.bytes())
```

Comment:
If this throws, we won't update the DB. We need to catch this, then maybe insert a NULL for replay ticks.

Reply:
Since you don't reraise the exception the next try-block will be walked through.
In the second except-block you don't update the db indeed.

Reply:
Always? Only if `get_replay_ticks` fails. All we really need to do is move `update_game_stats` out of the try block and set `ticks` in a similar way `replay_available` is.
I suppose handling each exception is a bit inelegant, but we really do want to catch and handle each call separately.

Reply:
Ah I didn't see that the replay ticks are calculated from the bytes in memory, expected it to re-read the file again for some reason. Now I understand what you're saying.